core/src/elements/fish_school.rs

```diff
@@ -302,3 +312,24 @@ pub fn update_fish_schools(
         }
     }
 }
+
+fn fish_school_kill_callback(entity: Entity) -> System {
+    (move |mut entities: ResMut<Entities>, mut fish_school: CompMut<FishSchool>| {
+        let mut to_kill: Vec<Entity> = Vec::new();
+        fish_school
+            .get_mut(entity)
+            .unwrap()
+            .fish
+            .iter()
+            .copied()
+            .for_each(|fish_entity| {
+                to_kill.push(fish_entity);
+            });
+        to_kill.push(entity);
+
+        to_kill.into_iter().for_each(|entity| {
+            entities.kill(entity);
+        });
```

Comment:
A minor tweak here can avoid an allocation. Often times this wouldn't work because iterating over entities usually needs to borrow `entities`, but in this case it's just a `Vec` so we're good.

```suggestion
        fish_school
            .get_mut(entity)
            .unwrap()
            .fish
            .iter()
            .copied()
            .for_each(|fish_entity| entities.kill(fish_entity));
```

Reply:
We could also kill the parent entity here with another call to `entities.kill(entity)`.